### PR TITLE
feat: upgrade to support storybook 7

### PR DESCRIPTION
**Pull request by taozhou-glean**
for https://github.com/storybookjs/addon-queryparams/issues/5

### CODE REVIEW

**Comment by taozhou-glean** [comment]
pretty much regenerated the addon with the latest https://github.com/storybookjs/addon-kit template

File: src/manager.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { addons } from "@storybook/manager-api";
+import { ADDON_ID } from "./constants";
+
+addons.register(ADDON_ID, () => {
```
Comment (taozhou-glean) [inline src/manager.ts:4]: keeping this for future changes

File: scripts/eject-typescript.mjs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+`);
+await $`yarn remove @babel/cli @babel/preset-typescript --ignore-scripts`;
+
+console.log(
+  chalk.green.bold`
+TypeScript Ejection complete!`,
+  chalk.green`
+Addon code converted with JS. The TypeScript build setup is still available in case you want to adopt TypeScript in the future.
+`
+);
```
Comment (ndelangen) [inline scripts/eject-typescript.mjs]: What's this code for? I see this PR is also using tsup 👍

Comment (taozhou-glean) [reply scripts/eject-typescript.mjs]: this is from the addon-kit template ;) https://github.com/storybookjs/addon-kit/tree/main/scripts I did not bother to change anything in the package.json other than meta info, but let me know if some of them should be removed

**Comment by taozhou-glean** [comment]
ping~ @ndelangen I would love to get this upgrade in soon to unblock our v7 upgrade, thanks

**Comment by taozhou-glean** [comment]
bump~

**Comment by thclark** [comment]
Would be amazing to get this merged!

**Comment by yannbf** [comment]
Hey everyone! So sorry this took so long. I'll be closing this PR as its work is superseded by #8.

**Pull request closed without merging**